app/src/main/java/com/example/farmeraid/home/add_edit_produce/AddEditProduceViewModel.kt

```diff
@@ -64,6 +64,34 @@ class AddEditProduceViewModel @Inject constructor(
         produceAmount.value = amount
     }
 
+    fun confirmDeleteProduce() {
+        snackbarDelegate.showSnackbar(
+            message = "Are you sure you want to delete ${produceName.value}",
+            actionLabel = "Yes",
+            onAction = { deleteProduce() }
+        )
+    }
+
+    private fun deleteProduce() {
+        viewModelScope.launch {
+            submitButtonUiState.value = submitButtonUiState.value.copy(isLoading = true)
+            val prodName = produceName.value
+            if (prodName == null) {
+                snackbarDelegate.showSnackbar("Invalid produce name")
+            } else if (initialProduceName != null) {
```

Comment:
Smart to use the initial produce name, incase the user modifies the name and tries to delete it afterwards 👍